File: Atividade9.py
def calcular_fatorial(numero):
    # Validar o número (não pode ser negativo)
    if numero < 0:
        print("Erro: O número não pode ser negativo!")
        return None  # Retorna None se for negativo

    # Cálculo do fatorial
    fatorial = 1
    for i in range(1, numero + 1):
        fatorial *= i

    # Retornar o fatorial
    return fatorial

# Solicitar o número ao usuário
numero = int(input("Digite um número para calcular o fatorial: "))

# Calcular e exibir o fatorial
resultado = calcular_fatorial(numero)
if resultado is not None:  # Verifica se o resultado não é None (erro)
    print(f"Fatorial de {numero}: {resultado}")

File: Atividade9.java
import java.util.Scanner;

public class Atividade9 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Solicitar o número
        System.out.print("Digite um número para calcular o fatorial: ");
        int numero = scanner.nextInt();

        // Validar o número (não pode ser negativo)
        if (numero < 0) {
            System.out.println("Erro: O número não pode ser negativo!");
            return; // Encerra o programa se for negativo
        }

        // Calcular e exibir o fatorial
        int fatorial = 1; // Fatorial inicia em 1
        for (int i = 1; i <= numero; i++) {
            fatorial *= i;
        }
        System.out.println("Fatorial de " + numero + ": " + fatorial);

        scanner.close();
    }
}
